apps/module-presale/contracts/Presale.sol

```diff
@@ -0,0 +1,336 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+import "@aragon/os/contracts/lib/math/SafeMath64.sol";
+
+import "@aragon/os/contracts/lib/token/ERC20.sol";
+import "@aragon/apps-shared-minime/contracts/MiniMeToken.sol";
+import "@aragon/apps-token-manager/contracts/TokenManager.sol";
+
+
+contract Presale is AragonApp {
+    using SafeMath for uint256;
+    using SafeMath64 for uint64;
+
+    /*
+     * Events
+     */
+
+    event SaleStarted();
+    event SaleClosed();
+    event TokensPurchased(address indexed buyer, uint256 daiSpent, uint256 tokensPurchased, uint256 purchaseId);
+    event TokensRefunded(address indexed buyer, uint256 daiRefunded, uint256 tokensBurned, uint256 purchaseId);
```

Comment:
Not sure about the rest of fundraising, but it seems odd to be DAI-specific here. Perhaps `contributionToken` would be better as a generic name?